crates/ghost_actor/src/ghost_actor.rs

```diff
@@ -271,19 +294,30 @@ pub trait GhostActor<'lt, P: GhostProtocol, A: GhostActor<'lt, P, A>>: Send + Sy
 }
 
 /// Slightly awkward helper class for obtaining an Endpoint for an actor's owner
-pub struct GhostInflator<'lt, P: GhostProtocol, A: 'lt + GhostActor<'lt, P, A>> {
+pub struct GhostInflator<
+    'lt,
+    P: GhostProtocol,
+    A: 'lt + GhostActor<'lt, P, A>,
+    S: GhostSystemRef<'lt>,
+> {
     finalize: Arc<GhostMutex<Option<GhostEndpointFullFinalizeCb<'lt, A>>>>,
-    sys_ref: GhostSystemRef<'lt>,
     sender: crossbeam_channel::Sender<(Option<RequestId>, P)>,
     receiver: crossbeam_channel::Receiver<(Option<RequestId>, P)>,
+    sys_ref: S,
 }
 
-impl<'lt, P: GhostProtocol, A: 'lt + GhostActor<'lt, P, A>> GhostInflator<'lt, P, A> {
+impl<
+        'lt,
+        P: GhostProtocol,
+        A: 'lt + GhostActor<'lt, P, A>,
+        S: 'lt + GhostSystemRef<'lt> + Sync + Send + Clone,
```

Comment:
```suggestion
        S: 'lt + GhostSystemRef<'lt>,
```

see previous suggested change, we *should* be able to get rid of the extra bounds here